### PR TITLE
wasmer: 0.17.0 -> 1.0.2

### PR DESCRIPTION
**Pull request by siriobalmelli**
Signed-off-by: Sirio Balmelli <sirio@b-ad.ch>

### CODE REVIEW

File: pkgs/development/interpreters/wasmer/default.nix [hunk 1]
```diff
@@ -8,25 +8,25 @@
...
+  cargoSha256 = "08r2b4s005w8r207jwq2fd43y3prgd8pg1m72aww1r7yrbxdr0v2";
 
   nativeBuildInputs = [ cmake pkg-config ];
 
   # Since wasmer 0.17 no backends are enabled by default. Backends are now detected
   # using the [makefile](https://github.com/wasmerio/wasmer/blob/master/Makefile).
   # Enabling cranelift as this used to be the old default. At least one backend is
   # needed for the run subcommand to work.
-  cargoBuildFlags = [ "--features" "backend-cranelift" ];
+  # cargoBuildFlags = [ "--features" "backend-cranelift" ];
```
Comment (Mic92) [inline pkgs/development/interpreters/wasmer/default.nix:29]: Does this not mean that no backends are enabled anymore?

Comment (siriobalmelli) [reply pkgs/development/interpreters/wasmer/default.nix:29]: `cargoBuildFlags = [ "--features" "backend-cranelift" ];` unfortunately fails with:

```
error: Package `wasmer-workspace v1.0.2 (/private/tmp/nix-build-wasmer-1.0.2.drv-0/source)` does not have the feature `backend-cranelift`
```

It appears back-ends are enabled by default; the documentation says to simply run `make build-wasmer`: https://docs.wasmer.io/ecosystem/wasmer/building-from-source.

The Makefile enables cranelift by default [here](https://github.com/wasmerio/wasmer/blob/9e0122e1f7bdb6fedf5fc17cc9a3a0fda30031a0/Makefile#L115).

This build can also be successfully run with `cargoBuildFlags = [ "--all-features" ];`, which may be the better approach?

Comment (SuperSandro2000) [reply pkgs/development/interpreters/wasmer/default.nix:29]: How big the overhead of just enabling all features?

**Comment by risicle** [comment]
Hmm. On macos 10.14, this appears to build fine but results in an empty output directory:

```
...
   Compiling wasmer-wast v1.0.2 (/private/tmp/nix-build-wasmer-1.0.2.drv-0/source/tests/lib/wast)
    Finished release [optimized] target(s) in 13m 57s
     Running target/x86_64-apple-darwin/release/deps/compilers-253ebc36b42b816b

running 0 tests

test result: ok. 0 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out; finished in 0.00s

     Running target/x86_64-apple-darwin/release/deps/deprecated-c82060b8c9746c44

running 0 tests

test result: ok. 0 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out; finished in 0.00s

Finished cargoCheckHook
@nix { "action": "setPhase", "phase": "installPhase" }
installing
Executing cargoInstallHook
Finished cargoInstallHook
@nix { "action": "setPhase", "phase": "fixupPhase" }
post-installation fixup
strip is /nix/store/s31f50385mj95q0ws54g2m3msxshvmkd-cctools-binutils-darwin-949.0.1/bin/strip
patching script interpreter paths in /nix/store/dnc6xwy245744rlbsiis2414ss2fqaq4-wasmer-1.0.2
```

**Comment by siriobalmelli** [comment]
I ran the Makefile from source in a nix-shell and the Makefile generates a `cargo` call with `--manifest-path lib/cli/Cargo.toml --bin wasmer`.

Adding just `--bin wasmer` is insufficient without also setting the manifest path.

Similarly, following https://github.com/wasmerio/wasmer/blob/master/Makefile#L22, seems that the sensible combination of back-end and compiler is *cranelift+JIT*.

@SuperSandro2000 ... enabling all back-ends and compilers gives some LLVM errors which I'd rather not get into right now.

File: pkgs/development/interpreters/wasmer/default.nix (hunk 1 above)
Comment (siriobalmelli) [reply pkgs/development/interpreters/wasmer/default.nix]: Good question - see comment below

**Pull request merged**